core/src/main/scala/com/raphtory/api/analysis/graphstate/Counter.scala

```diff
@@ -0,0 +1,32 @@
+package com.raphtory.api.analysis.graphstate
+
+import scala.collection.mutable
+
+/**
+  * Public interface for the Counter API within [[GraphState]].
+  *
+  *  A Counter maintains the distribution of categorical groupings such as community, connected component or node type.
+  *  This enables measures such as the size/relative size of the largest group or largest k groups, filtering to include
+  *  only the largest/largest k groups etc.
+  *
+  *  @tparam T Value type of the category being grouped (usually long if referring to a group or component id)
+  *
+  * @see
+  * [[GraphState]], [[Histogram]]
+  */
+
+abstract class Counter[T] {
+
+  /** Total of the counts */
+  def totalCount : Int
+
+  /** Getter function for the map of counts */
+  def getCounts : mutable.Map[T,Int]
```

Comment:
I would make this return a MapView to encapsulate the mutable state otherwise this can be updated externally bypassing the synchronized